src/lammpsio/topology.py

```diff
@@ -241,3 +259,64 @@ class Impropers(Topology):
 
     def __init__(self, N, num_types=None):
         super().__init__(N=N, num_members=4, num_types=num_types)
+
+
+class LabelMap(collections.abc.MutableMapping):
+    """Label map between typeids and types.
+
+    Parameters
+    ----------
+    map : dict
+        Map of typeids to types.
+
+    Attributes
+    ----------
+    types : tuple
+        Types in label map.
+    typeid : tuple
+        Typeids in label map.
+    """
+
+    def __init__(self, map=None):
+        self._map = {}
+        if map is not None:
+            self.update(map)
+
+    def __getitem__(self, key):
+        return self._map[key]
+
+    def __setitem__(self, key, value):
+        self._map[key] = value
+
+    def __delitem__(self, key):
+        del self._map[key]
+
+    def __iter__(self):
+        return iter(self._map)
+
+    def __len__(self):
+        return len(self._map)
+
+    @property
+    def types(self):
+        """Types in label map.
+
+        Returns
+        -------
+        tuple
+            Label types.
+
+        """
```

Comment:
Properties don't usually get a separate return since they are simple. It should go like this:
```suggestion
        """tuple: Types in label map."""
```